scala-thoughts.md

```diff
@@ -0,0 +1,32 @@
+## General Reflections
+
+After completing the assignment, I feel as though I have a much better grasp of Scala. I have never paid quite so much attention to the process of learning a new language. When we create our own DSLs, hopefully this awareness will make it easier for us to see things from the perspective of the user. I certainly put a lot more thought into how I used various Scala resources.
+
+
+## Helpful Tools
+
+Throughout the assignment, the Scala documentation was my most consulted resource. I found the documentation to be thorough, especially the section on collections. It especially came in handy for the list comprehension exercises in the first lab. I did not have much background with list comprehensions from previous computer science classes, so the documentation was especially useful in showing me what functions were already available in Scala.
+
+The other resource that I ended up using quite a bit was Stack Overflow. Whenever I had a question about how to do something slightly unclear in Scala, I could almost always find someone who had a similar problem on Stack Overflow. This was especially useful if I got errors that were unclear and I couldn't find the specific issue in the documentation. I also found the links to pattern matching articles in the assignment to be very helpful. 
+
```

Comment:
Definitely agree on Stack Overflow being useful. I didn't get as much out of the documentation though--I found it a bit hard to navigate.